packages/core/admin/admin/src/components/LanguageProvider/reducer.js

```diff
@@ -4,47 +4,32 @@
  *
  */
 
-import { get, includes, split } from 'lodash';
-
-// Import supported languages from the translations folder
-import trads from '../../translations';
-import { CHANGE_LOCALE } from './constants';
-
-const languages = Object.keys(trads);
-
-// Define a key to store and get user preferences in local storage.
 const localStorageKey = 'strapi-admin-language';
 
-// Detect user language.
-const userLanguage =
-  window.localStorage.getItem(localStorageKey) ||
-  window.navigator.language ||
-  window.navigator.userLanguage;
-
-let foundLanguage = includes(languages, userLanguage) && userLanguage;
-
-if (!foundLanguage) {
-  // Split user language in a correct format.
-  const userLanguageShort = get(split(userLanguage, '-'), '0');
-
-  // Check that the language is included in the admin configuration.
-  foundLanguage = includes(languages, userLanguageShort) && userLanguageShort;
-}
-
 const initialState = {
-  locale: foundLanguage || 'en',
+  localesNativeNames: { en: 'English' },
+  locale: 'en',
 };
 
-function languageProviderReducer(state = initialState, action) {
+const languageProviderReducer = (state = initialState, action) => {
   switch (action.type) {
-    case CHANGE_LOCALE:
+    case 'CHANGE_LOCALE': {
+      const { locale } = action;
+
+      if (!state.localesNativeNames[locale]) {
+        return state;
+      }
+
       // Set user language in local storage.
-      window.localStorage.setItem(localStorageKey, action.locale);
+      window.localStorage.setItem(localStorageKey, locale);
```

Comment:
I think that it's not the place for synchronizing with the local storage. I would suggest to put it in an effect but not inside the reducer

Reply:
What about it in the handler?